### PR TITLE
Authentication via J-PAKE

### PR DESCRIPTION
**Pull request by maxyakimakha**
CDDL message definitions for J-PAKE authentication protocol. The protocol is completely symmetric, both sides can use the same message format. Fields are named from the message sender perspective.

***
<a href="https://pr-preview.s3.amazonaws.com/maxyakimakha/openscreenprotocol/pull/164.html" title="Last updated on May 20, 2019, 6:27 PM UTC (5fddd56)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webscreens/openscreenprotocol/164/0580952...maxyakimakha:5fddd56.html" title="Last updated on May 20, 2019, 6:27 PM UTC (5fddd56)">Diff</a>

### CODE REVIEW

**Comment by maxyakimakha** [comment]
Tentative PR, awaiting decision on authentication method we choose to go with.

File: index.bs [hunk 1]
```diff
@@ -516,6 +516,51 @@ Note: the values of 32 above (for salt length, keyLength) are based on the
...
+4. HKDF with SHA-256 as KDF function
+5. The second key confirmation method described in https://tools.ietf.org/html/rfc8236#section-5
+
+An agent may send auth-jpake-round1 message at any time to start
+the authentication process with the other agent.
+
+After sending auth-jpake-round1, the agent expects to receive auth-jpake-round1
+and auth-jpake-round2 messages.
+
+After receiving auth-jpake-round1 message, the agent validates the message and, if
```
Comment (pthatcherg) [inline index.bs]: After receiving a auth-jpake-round1 message,

File: index.bs [hunk 2]
```diff
@@ -516,6 +516,51 @@ Note: the values of 32 above (for salt length, keyLength) are based on the
...
+the validation passes, sends:
+1. auth-jpake-round1 message, in case the agent has not sent it yet;
+2. auth-jpake-round2 message.
+
+After sending auth-jpake-round2, the agent expects to receive
+auth-jpake-confirmation message.
```
Comment (pthatcherg) [inline index.bs]: After receiving a auth-jpake-round2 message,

Comment (pthatcherg) [reply index.bs]: (oops, below, not above)

File: index.bs [hunk 3]
```diff
@@ -516,6 +516,51 @@ Note: the values of 32 above (for salt length, keyLength) are based on the
...
+
+After receiving auth-jpake-round2 message, the agent validates the message and,
+if the validation passes and the agent has received auth-jpake-round1 message
+as well, the agent sends auth-jpake-confirmation message.
+
+After receiving auth-jpake-confirmation message, the agent verifies MAC to confirm
```
Comment (pthatcherg) [inline index.bs]: verifies the MAC

**Pull request closed without merging**